source/MRViewer/MRSaveOnClose.cpp

```diff
@@ -88,24 +89,27 @@ void SaveOnClosePlugin::preDraw_()
         {
             auto savePath = SceneRoot::getScenePath();
             if ( savePath.empty() )
-                savePath = saveFileDialog( { {}, {},SceneFileFilters } );
+                savePath = saveFileDialog( { {}, {}, SceneFileWriteFilters } );
 
-            ProgressBar::orderWithMainThreadPostProcessing( "Saving scene", [savePath, &root = SceneRoot::get(), viewer = Viewer::instance()]()->std::function<void()>
-            {
-                auto res = serializeObjectTree( root, savePath, ProgressBar::callBackSetProgress );
-                if ( !res.has_value() )
-                    spdlog::error( res.error() );
-
-                return[savePath, viewer, success = res.has_value()]()
+            ImGui::CloseCurrentPopup();
```

Comment:
why this is here?

Reply:
- Close application
- Application close dialog: Save, Don't Save, Cancel?
- Press Save but then dismiss the Save As dialog with Cancel
- The application close dialog should close as well